wagtail/wagtailcore/models.py

```diff
@@ -242,6 +242,11 @@ def specific(self):
         return self.get_queryset().specific()
 
 
+class ProxyPageManager(PageManager):
+    def get_queryset(self):
+        return super().get_queryset().type(self.model)
```

Comment:
Calling `.type(self.model)` is not enough here because that will return all the objects for the concrete model. This does work however:
`return super(ProxyPageManager, self).get_queryset().filter(content_type=ContentType.objects.get_for_model(self.model, for_concrete_model=False))`

Or even `PageQuerySet.type` could take a `for_concrete_models=False` argument too, which would make it return proxy objects instead of concrete ones. And I wonder if returning proxy objects *by default* would make sense: I think that's what people would expect but might have some effects on some internal wagtail code (and it's not what django does by default either).